manager/controlapi/service.go

```diff
@@ -238,6 +286,10 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 		return nil, err
 	}
 
+	if err := s.checkPortConflicts(request.Spec); err != nil {
+		return nil, err
+	}
```

Comment:
Let's either make `CreateService` transactional or move the `checkPortConflicts` check out of `s.store.Update`, so they are consistent. I think I prefer the latter, but there's no obvious right answer.